docker-compose.yaml

```diff
@@ -6,15 +6,21 @@ services:
     container_name: wdio-cucumber
     links:
       - selenium
+    volumes:
+      - ./src/:/usr/src/app/src
+      - ./test/:/usr/src/app/test
+      - ./babel.config.js:/usr/src/app/babel.config.js
+      - ./wdio.conf.js:/usr/src/app/wdio.conf.js
+      - ./wdio.BUILD.conf.js:/usr/src/app/wdio.BUILD.conf.js
 
   selenium:
-    #image: selenium/standalone-chrome-debug:3.7.1
-    image: selenium/standalone-firefox-debug:3.7.1
+    image: selenium/standalone-chrome-debug
+    # image: selenium/standalone-firefox-debug
     expose:
       - "4444"
     privileged: true
 
   hub:
-    image: selenium/hub:3.7.1
+    image: selenium/hub
     ports:
       - "4444:4444"
```

Comment:
Now that this is working on containers, it would be good to refactor the docker-compose files to be more efficient.  For example when running in a pipeline you wouldn't need all of the volumes of the build server bound to the container.  You also don't need to expose the ports as that could lead to conflicts if you have multiple builds running.  For example two acceptance tests for different services run in Jenkins at the same time, both want to expose port 4444 so you have a conflict.

To resolve this you could change it to the below.

## docker-compose.yaml
```
version: "3.7"
services:
  wdio-cucumber:
    build: .
    image: wdio-cucumber
    container_name: wdio-cucumber
    depends_on:
      - selenium

  selenium:
    image: selenium/standalone-chrome-debug
    privileged: true

  hub:
    image: selenium/hub
```
## docker-compose.override.yaml
```
version: "3.7"
services:
  wdio-cucumber:
    volumes:
      - ./src/:/usr/src/app/src
      - ./test/:/usr/src/app/test
      - ./babel.config.js:/usr/src/app/babel.config.js
      - ./wdio.conf.js:/usr/src/app/wdio.conf.js
      - ./wdio.BUILD.conf.js:/usr/src/app/wdio.BUILD.conf.js

  hub:
    ports:
      - "4444:4444"
```

When running `docker-compose` commands such as `docker-compose up` you can specify additional files to apply to the command with the `-f` flag.  These files are applied in the order they are listed in the command and will add to/overwrite any matching properties of previous files.

So for example you could do `docker-compose -f docker-compose.yaml -f docker-compose.development.yaml up`

One nice feature of Docker compose is that if you create a file called `docker-compose.override.yaml` it will automatically be used without you needing to specify it.

So `docker-compose -f docker-compose.yaml -f docker-compose.override.yaml up` is actually the same as `docker-compose up`.  So splitting them is no more of a burden for local dev.

When it comes to the build server, we don't want to use the override file so instead we can specify we just want the base file with `docker-compose -f docker-compose.yaml up`

Reply:
Right, thanks for these tips, actioned this in next commits